### PR TITLE
Bugfix: markers with no valid reads (no coverage)

### PR DESCRIPTION
**Pull request by standage**
Certain portions of the MicroHapulator code were implemented with an implicit assumption that there would be at least one valid read for each marker (valid = spanning all corresponding SNPs, and thus not discarded). This PR updates the code to work correctly even if one or more markers has no valid reads.

----------

- [x] Changes are clearly described above
- ~~Any relevant issue threads are referenced in the description~~
- [x] Any new features are tested (see [docs/DEVEL.md](docs/DEVEL.md) for details)
- [x] Substantial changes are documented in CHANGELOG.md (see https://keepachangelog.com/en/1.0.0/)

### CODE REVIEW

File: microhapulator/tests/test_type.py [hunk 1]
```diff
@@ -151,3 +151,11 @@ def test_type_no_var_offsets():
     message = r"marker IDs unique to set1={mh01KK-205, mh02KK-005, mh03KK-006};"
     with pytest.raises(ValueError, match=message):
         mhapi.type(bam, tsv)
+
+
+def test_type_marker_with_no_coverage():
+    bam = data_file("bam/nocov.bam")
+    tsv = data_file("def/nocov-offsets.tsv")
```
Comment (standage) [inline microhapulator/tests/test_type.py:158]: For testing purposes, constructed a BAM file with alignments to four markers.

File: microhapulator/tests/data/def/nocov-offsets.tsv [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Marker	Offset
+mh02USC-2qE	151
+mh02USC-2qE	170
+mh02USC-2qE	174
+mh03USC-3qB	143
```
Comment (standage) [inline microhapulator/tests/data/def/nocov-offsets.tsv:5]: The four markers in the BAM file are a subset of the 5 markers in this definition file.

File: microhapulator/profile.py [hunk 3]
```diff
@@ -322,6 +322,7 @@ def __init__(self, fromfile=None):
 
     def record_coverage(self, marker, cov_by_pos, ndiscarded=0):
         self.data["markers"][marker] = {
+            "genotype": list(),
```
Comment (standage) [inline microhapulator/profile.py:325]: Fix.

**Review by RyanBerger98 — APPROVED** [review]
Looks good! I can confirm that the tests pass on my end and the results are what you would expect.

**Pull request merged**